internal/oci/annotations.go

```diff
@@ -9,8 +9,9 @@ import (
 const (
 	SpecName = "remoteproc.name"
 
-	StateResolvedPath = "remoteproc.resolved-path"
-	StateFirmware     = "remoteproc.firmware"
+	StateDriverPath         = "remoteproc.driver-path"
+	StateFirmwarePath       = "remoteproc.firmware-path"
+	StateStoredFirmwareName = "remoteproc.stored-firmware-name"
```

Comment:
I'm now realising that this is optional entry and I think we should mark it as such. 
We can't do the usual validation for presence in `validateStateAnnotations`, because it'll be absent until `start` is invoked.

Now, I'd keep the indication of "optionality" in the codebase but not expose that in the key we write to the `state.json` file. Something like:

```
OptionalStateStoredFirmwareName = "remoteproc.stored-firmware-name"
```

Hopefully, that then makes it explicit in call sites, that we need to check if this key has value.